test/build/base-url/base-url.test.js

```diff
@@ -1,15 +1,20 @@
-const fs = require('fs');
 const path = require('path');
 const cheerio = require('cheerio');
+const {setupBuildTest, readFiles} = require('../../test-utils');
 
 const cwd = path.join(__dirname, 'build');
 
-const html = fs.readFileSync(path.join(cwd, 'index.html'), 'utf8');
-const indexJS = fs.readFileSync(path.join(cwd, 'index.js'), 'utf-8');
-const distJS = fs.readFileSync(path.join(cwd, '_dist_', 'index.js'), 'utf-8');
-const $ = cheerio.load(html);
+let files = {};
+let $;
 
 describe('buildOptions.baseUrl', () => {
+  beforeAll(() => {
+    setupBuildTest(__dirname);
+
+    files = readFiles(['index.html', 'index.js', '_dist_/index.js'], {cwd});
```

Comment:
also, comment as I'm writing my first test: do we need to maintain an allowlist like this? I'm seeing some unnecessary overhead managing this array and the different `files[PATH]` in the tests below. Since there's no connection between the two, I could see this getting out of date as well.

A better interface imo would be `readFile(path)`, which looks up the file during the test itself:

```
it('...', (async () => {
  expect(await getFile('/index.js')).toEqual(...);
});
```

Reply:
You‘re right that the initialization is awkward. But reading file contents is the bulk of these tests, and most tests can‘t have an inline `expect(await getFile(…))` like in your example. Most tests reuse file contents across tests, so it needs to exist in some shared variable.

In writing tests I found myself even getting confused by having to name every file in a project (how do I name `/index.js` vs `/_dist_/index.js` vs `/_dist_/components/foo2.js`?). I found it much easier to just refer to the filepath itself, both in reading & writing tests. I think even if you scope variables within tests, you still have the same problem of having to come up with names for all these files.

Re-reading files isn‘t really a big deal; I’m mostly concerned about the DX.

But that said, WDYT about a glob pattern? e.g. `readFiles(['**/*.js', '**/*.html'])`. Or even shorter: `readFiles(someDir, {ignore: […]})` to read all files? That’d fix the allowlist problem and still save having to name dozens of files which was my main aim.

Reply:
I'm not ready to give up on the idea of solving this with a getter function idea that easily :) Mainly because the objective overhead cost of having to maintain that duplicate file allowlist always matching the files referenced in the individual tests doesn't beat out the subjective "easier variable naming" problem.

I think your issue is addressable here. For example, I'd be happy with either of these:

```js
// 1.
const fileToTest = await getFile('/index.js');
expect(fileToTest).toInclude(`import __SNOWPACK_ENV__ from './__snowpack__/env.js';`);
expect(fileToTest).toInclude(`import.meta.env = __SNOWPACK_ENV__;`);

// 2. (with the result memoized so that only one lookup is actually used)
expect(await getFile('/index.js')).toInclude(`import __SNOWPACK_ENV__ from './__snowpack__/env.js';`);
expect(await getFile('/index.js')).toInclude(`import.meta.env = __SNOWPACK_ENV__;`);
```

BUT... if you're still on board with moving towards running these tests through the JS Build API as a part of our v3 release, then I'm fine with the existing `files: []` pattern for now, since the helper that I'm proposing would most likely go away when that API arrives.

Reply:
> BUT... if you're still on board with moving towards running these tests through the JS Build API as a part of our v3 release, then I'm fine with the existing `files: []` pattern for now, since the helper that I'm proposing would most likely go away when that API arrives.

Ah yeah I’d be down for that! I think that’d be a great replacement for the current `files: []` pattern.

> I think your issue is addressable here. For example, I'd be happy with either of these:

Maybe I didn’t explain clearly… the current situation is more like imagine if #1 had 20+ files to keep track of, and needed to be referenced.

#2 could work, but in order to be memoized we’d really need to beef it up and I’d rather just discard it in favor of the Node API later on.

See https://github.com/snowpackjs/snowpack/pull/1861/files#diff-5097166b779657484196ba597acf4521937f197da2c8803d52480c87d1fb8c75R13 for a reference on the changes… for now I changed it so that it just reads a directory.